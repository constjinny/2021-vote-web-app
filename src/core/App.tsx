import { ReactElement, useEffect } from "react";
import styled from "@emotion/styled";
import { CommAPI } from "../api/sample.api";

import Routes from "../routes/Routes";

function App(): ReactElement {
  useEffect(() => {
    // NOTI: 임시 API사용으로 최초 진입 시 세션에 데이터 별도 세팅
    CommAPI.setInitData();
  }, []);

  return (
    <WrapStyle>
      <Routes />
    </WrapStyle>
  );
}

export default App;

const WrapStyle = styled.div`
  height: 100%;
`;
