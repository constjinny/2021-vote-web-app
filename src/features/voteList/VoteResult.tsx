import { ReactElement } from "react";
import styled from "@emotion/styled";

import { IVoteItemProps, IVoteOptionProps } from "./type";

export function VoteResult({ data }: { data: IVoteItemProps }): ReactElement {
  const { voteOptions, isResultId } = data;

  return (
    <VoteDetailStyle>
      <OptionListStyle>
        {voteOptions.map((option: IVoteOptionProps, index: number) => {
          const { optionId, optionValue, selectedIds, count } = option;
          const isResult = isResultId === optionId;
          const SelectedIdList = selectedIds.map((id: string) => (
            <span key={id}>{id}</span>
          ));

          return (
            <OptionStyle key={optionId} isResult={isResult}>
              <span>{index + 1}</span>

              <div>
                {optionValue}
                <span>{`${count} ëª…`}</span>
              </div>

              {SelectedIdList && (
                <SelectedIdsStyle>{SelectedIdList}</SelectedIdsStyle>
              )}
            </OptionStyle>
          );
        })}
      </OptionListStyle>
    </VoteDetailStyle>
  );
}

const VoteDetailStyle = styled.div`
  margin-top: 20px;
  padding: 20px;
  border-radius: 5px;
  border: 1px solid #cacaca;
  box-sizing: border-box;
`;

const OptionListStyle = styled.ul`
  > p {
    margin-bottom: 10px;
  }
`;

const OptionStyle = styled.li<{ isResult: boolean }>`
  overflow: hidden;
  position: relative;
  margin-top: 10px;
  padding: 10px 20px 10px 50px;
  font-size: 14px;
  border-radius: 5px;
  border: 1px solid #cacaca;
  ${({ isResult }) => isResult && "border-color: #5ea9e7;"};
  > span {
    position: absolute;
    top: 10px;
    left: 20px;
    width: 20px;
    height: 20px;
    text-align: center;
    font-weight: bold;
    line-height: 20px;
    border-radius: 50%;
    background-color: ${({ isResult }) => (isResult ? "#5ea9e7" : "#cacaca")};
    color: #fff;
  }
  > div {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
`;

const SelectedIdsStyle = styled.div`
  display: flex;
  flex-wrap: wrap;
  font-size: 14px;
  > span {
    margin: 5px 2px;
    padding: 1px 5px;
    border-radius: 5px;
    background-color: #cacaca;
    color: #fff;
  }
`;
