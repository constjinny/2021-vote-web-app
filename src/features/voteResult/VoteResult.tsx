import { ReactElement } from "react";
import styled from "@emotion/styled";

export function VoteResult(): ReactElement {
  return <VoteResultStyle>VoteResult</VoteResultStyle>;
}

const VoteResultStyle = styled.div``;
