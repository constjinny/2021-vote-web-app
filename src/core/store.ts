import { configureStore, Action } from "@reduxjs/toolkit";
import { combineReducers } from "@reduxjs/toolkit";
import { ThunkAction } from "redux-thunk";
import { useDispatch } from "react-redux";

import { voteListReducer } from "../features/voteList";
import { voteWriteReducer } from "../features/voteCreate";

export const rootReducer = combineReducers({
  voteListReducer,
  voteWriteReducer,
});

export type IRootState = ReturnType<typeof rootReducer>;

const store = configureStore({
  reducer: rootReducer,
});

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export type AppThunk = ThunkAction<void, IRootState, null, Action<string>>;

export default store;
