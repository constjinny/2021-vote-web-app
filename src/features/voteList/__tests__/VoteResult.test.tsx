import { configureStore } from "@reduxjs/toolkit";
import { rootReducer } from "../../../core/store";
import { Provider } from "react-redux";
import { shallow } from "enzyme";
import { VoteResult } from "../VoteResult";

const testData = {
  author: {
    id: "jinny",
    nickName: "jinny",
    profileUrl: "jinny_profile.png",
  },
  closeDate: "2021년 6월 19일",
  isClose: true,
  isEdit: true,
  isOpen: true,
  isResultId: "",
  isSelectId: "",
  openDate: "2021년 6월 18일",
  voteId: "b830191fdi",
  voteName: "테스트 투표",
  voteOptions: [
    {
      count: 0,
      optionId: "b830191fdi-item1",
      optionValue: "투표 항목",
      selectedIds: [],
    },
  ],
};

describe("VoteResult", () => {
  const store = configureStore({
    reducer: rootReducer,
  });
  let component: any = null;

  it("renders VoteResult component", () => {
    component = shallow(
      <Provider store={store}>
        <VoteResult data={testData} />
      </Provider>
    );
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
