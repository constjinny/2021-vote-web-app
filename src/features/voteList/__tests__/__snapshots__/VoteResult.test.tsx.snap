// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteResult matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <VoteResult
    data={
      Object {
        "author": Object {
          "id": "jinny",
          "nickName": "jinny",
          "profileUrl": "jinny_profile.png",
        },
        "closeDate": "2021년 6월 19일",
        "isClose": true,
        "isEdit": true,
        "isOpen": true,
        "isResultId": "",
        "isSelectId": "",
        "openDate": "2021년 6월 18일",
        "voteId": "b830191fdi",
        "voteName": "테스트 투표",
        "voteOptions": Array [
          Object {
            "count": 0,
            "optionId": "b830191fdi-item1",
            "optionValue": "투표 항목",
            "selectedIds": Array [],
          },
        ],
      }
    }
  />
</ContextProvider>
`;
